android/quest/src/main/java/org/smartregister/fhircore/quest/util/extensions/ContextExtensions.kt

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2021-2024 Ona Systems, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.smartregister.fhircore.quest.util.extensions
+
+import android.content.Context
+import android.content.ContextWrapper
+import androidx.appcompat.app.AppCompatActivity
+
+fun Context.getActivity(): AppCompatActivity? {
+  var currentContext = this
+  while (currentContext is ContextWrapper) {
+    if (currentContext is AppCompatActivity) {
+      return currentContext
+    }
+    currentContext = currentContext.baseContext
+  }
+  return null
+}
```

Comment:
I understand why this was added. You can re-use a similar implementation from `org.smartregister.fhircore.engine.util.extension.AndroidExtensionsKt#getActivity`